From 0d98548c16bc2fbaa2f3bba09b0aec4a2fa0da2c Mon Sep 17 00:00:00 2001
From: Marian Florea <marian.florea@windriver.com>
Date: Wed, 17 Oct 2018 11:46:24 +0000
Subject: [PATCH 11/11] arm64: dts: stratix10: use clock bindings for the
 Stratix10 platform

Reverts 56594bb5408396a53a0794f22a31ea98fc999299
(FogBugz #488843-1: dts: clock bindings for the SoCFPGA Stratix10)

and

Backports
commit d93101abe41e9596555a0f7f6f775e543b71c441 upstream

Use the clock bindings for the Stratix10 SoC. This includes changing the
old binding of "intc,clk-s10-mgr" to "intel,stratix10-clkmgr". The reason that
this can be done is that there are currently no clock driver for Stratix10,
thus there are no consumers of the old binding. So changing the binding will
not break any legacy code.

Signed-off-by: Dinh Nguyen <dinguyen@kernel.org>
Signed-off-by: Marian Florea <marian.florea@windriver.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi  |  534 ++------------------
 .../boot/dts/altera/socfpga_stratix10_socdk.dts    |    8 +-
 2 files changed, 53 insertions(+), 489 deletions(-)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
index dd266d5..7d7a30d 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
@@ -17,6 +17,7 @@
 /dts-v1/;
 #include <dt-bindings/reset/altr,rst-mgr-s10.h>
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/clock/stratix10-clock.h>
 
 / {
 	compatible = "altr,socfpga-stratix10";
@@ -113,474 +114,38 @@
 			fpga-mgr = <&fpga_mgr>;
 		};
 
-		clkmgr@ffd10000 {
-			compatible = "altr,clk-mgr";
+		clkmgr: clock-controller@ffd10000 {
+			compatible = "intel,stratix10-clkmgr";
 			reg = <0xffd10000 0x1000>;
+			#clock-cells = <1>;
+		};
 
-			clocks {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				cb_intosc_hs_div2_clk: cb_intosc_hs_div2_clk {
-					#clock-cells = <0>;
-					compatible = "fixed-clock";
-				};
-
-				cb_intosc_ls_clk: cb_intosc_ls_clk {
-					#clock-cells = <0>;
-					compatible = "fixed-clock";
-				};
-
-				f2s_free_clk: f2s_free_clk {
-					#clock-cells = <0>;
-					compatible = "fixed-clock";
-				};
-
-				osc1: osc1 {
-					#clock-cells = <0>;
-					compatible = "fixed-clock";
-				};
-
-				main_pll: main_pll@44 {
-					#address-cells = <1>;
-					#size-cells = <0>;
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-pll-clock";
-					clocks = <&osc1>, <&cb_intosc_hs_div2_clk>,
-						 <&f2s_free_clk>;
-					reg = <0x74>;
-
-					main_mpu_base_clk: main_mpu_base_clk {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						div-reg = <0x84 0 8>;
-					};
-
-					main_noc_base_clk: main_noc_base_clk {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						div-reg = <0x88 0 8>;
-					};
-
-					main_emaca_clk: main_emaca_clk@68 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x50>;
-					};
-
-					main_emacb_clk: main_emacb_clk@6c {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x54>;
-					};
-
-					main_emac_ptp_clk: main_emac_ptp_clk@70 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x58>;
-					};
-
-					main_gpio_db_clk: main_gpio_db_clk@74 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x5c>;
-					};
-
-					main_sdmmc_clk: main_sdmmc_clk@78 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk"
-;
-						clocks = <&main_pll>;
-						reg = <0x60>;
-					};
-
-					main_s2f_usr0_clk: main_s2f_usr0_clk@7c {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x64>;
-					};
-
-					main_s2f_usr1_clk: main_s2f_usr1_clk@80 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x68>;
-					};
-
-					main_psi_ref_clk: main_psi_ref_clk@84 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>;
-						reg = <0x6c>;
-					};
-				};
-
-				periph_pll: periph_pll@e4 {
-					#address-cells = <1>;
-					#size-cells = <0>;
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-pll-clock";
-					clocks = <&osc1>, <&cb_intosc_hs_div2_clk>,
-						 <&f2s_free_clk>;
-					reg = <0xe4>;
-
-					peri_mpu_base_clk: peri_mpu_base_clk {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&periph_pll>;
-						reg = <0xf4 0 8>;
-					};
-
-					peri_noc_base_clk: peri_noc_base_clk {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&periph_pll>;
-						reg = <0xf8 0 8>;
-					};
-
-					peri_emaca_clk: peri_emaca_clk@e8 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xbc>;
-					};
-
-					peri_emacb_clk: peri_emacb_clk@ec {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xc0>;
-					};
-
-					peri_emac_ptp_clk: peri_emac_ptp_clk@f0 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xc4>;
-					};
-
-					peri_gpio_db_clk: peri_gpio_db_clk@f4 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xc8>;
-					};
-
-					peri_sdmmc_clk: peri_sdmmc_clk@f8 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xcc>;
-					};
-
-					peri_s2f_usr0_clk: peri_s2f_usr0_clk@fc {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&periph_pll>;
-						reg = <0xd0>;
-					};
-
-					peri_s2f_usr1_clk: peri_s2f_usr1_clk@100 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xd4>;
-					};
-
-					peri_psi_ref_clk: peri_psi_ref_clk@104 {
-						#clock-cells = <0>;
-						compatible = "altr,socfpga-s10-perip-clk";
-						clocks = <&main_pll>, <&periph_pll>,
-							 <&osc1>, <&cb_intosc_hs_div2_clk>,							    <&f2s_free_clk>;
-						reg = <0xd8>;
-					};
-				};
-
-				boot_clk: boot_clk@0 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&osc1>, <&cb_intosc_hs_div2_clk>;
-					reg = <0x0>;
-				};
-
-				mpu_free_clk: mpu_free_clk@48 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&main_mpu_base_clk>, <&peri_mpu_base_clk>,
-						 <&osc1>, <&cb_intosc_hs_div2_clk>,
-						 <&f2s_free_clk>;
-					reg = <0x48>;
-				};
-
-				noc_free_clk: noc_free_clk@4c {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&main_noc_base_clk>, <&peri_noc_base_clk>,
-						 <&osc1>, <&cb_intosc_hs_div2_clk>,
-						 <&f2s_free_clk>;
-					reg = <0x4c>;
-				};
-
-				s2f_user0_free_clk: s2f_user0_free_clk@104 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&main_s2f_usr0_clk>, <&peri_s2f_usr0_clk>,
-						 <&osc1>, <&cb_intosc_hs_div2_clk>,
-						 <&f2s_free_clk>;
-					reg = <0x64>;
-				};
-
-				l4_sys_free_clk: l4_sys_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&noc_free_clk>;
-					fixed-divider = <4>;
-				};
-
-				noc_clk: noc_clk@30 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&noc_free_clk>, <&boot_clk>;
-					bypass-reg = <0x3c 1>;
-				};
-
-				emaca_free_clk: emaca_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&main_emaca_clk>;
-					bypass-reg = <0xb0 0>;
-				};
-
-				emacb_free_clk: emacb_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&main_emacb_clk>;
-					bypass-reg = <0xb0 1>;
-				};
-
-				emac_ptp_free_clk: emac_ptp_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&peri_emac_ptp_clk>;
-					bypass-reg = <0xb0 2>;
-				};
-
-				gpio_db_free_clk: gpio_db_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&peri_gpio_db_clk>;
-					bypass-reg = <0xb0 3>;
-				};
-
-				sdmmc_free_clk: sdmmc_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&peri_sdmmc_clk>;
-					bypass-reg = <0xb0 4>;
-				};
-
-				s2f_user1_free_clk: s2f_user1_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&peri_s2f_usr1_clk>;
-					bypass-reg = <0xb0 5>;
-				};
-
-				psi_ref_free_clk: psi_ref_free_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-perip-clk";
-					clocks = <&boot_clk>, <&peri_psi_ref_clk>;
-					bypass-reg = <0xb0 6>;
-				};
-
-				mpu_clk: mpu_clk@30 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&boot_clk>, <&mpu_free_clk>;
-					bypass-reg = <0x3c 0>;
-					clk-gate = <0x30 0>;
-				};
-
-				mpu_periph_clk: mpu_periph_clk@30 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&mpu_clk>;
-					fixed-divider = <4>;
-					clk-gate = <0x30 0>;
-				};
-
-				mpu_l2ram_clk: mpu_l2ram_clk@30 {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&mpu_clk>;
-					fixed-divider = <2>;
-					clk-gate = <0x30 0>;
-				};
-
-				l4_main_clk: l4_main_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&noc_clk>;
-					div-reg = <0x70 0 2>;
-					clk-gate = <0x30 1>;
-				};
-
-				l4_mp_clk: l4_mp_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&noc_clk>;
-					div-reg= <0x70 8 2>;
-					clk-gate = <0x30 2>;
-				};
-
-				l4_sp_clk: l4_sp_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&noc_clk>;
-					div-reg= <0x70 16 2>;
-					clk-gate = <0x30 3>;
-				};
-
-				cs_at_clk: cs_at_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&noc_clk>;
-					div-reg= <0x70 24 2>;
-					clk-gate = <0x30 4>;
-				};
-
-				cs_trace_clk: cs_trace_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&noc_clk>;
-					div-reg= <0x70 26 2>;
-					clk-gate = <0x30 4>;
-				};
-
-				cs_pdbg_clk: cs_pdbg_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&cs_at_clk>;
-					div-reg= <0x70 28 1>;
-					clk-gate = <0x30 4>;
-				};
-
-				cs_timer_clk: cs_timer_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&noc_clk>;
-					clk-gate = <0x30 5>;
-				};
-
-				s2f_user0_clk: s2f_user0_clk@3c {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&boot_clk>, <&f2s_free_clk>;
-					bypass-reg = <0x3c 2>;
-					clk-gate = <0x30 6>;
-				};
-
-				emac0_clk: emac0_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&peri_emaca_clk>;
-					fixed-divider = <4>;
-					clk-gate = <0xa4 0>;
-				};
-
-				emac1_clk: emac1_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&emaca_free_clk>, <&emacb_free_clk>;
-					bypass-reg = <0xdc 27>;
-					clk-gate = <0xa4 1>;
-				};
-
-				emac2_clk: emac2_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&emaca_free_clk>, <&emacb_free_clk>;
-					bypass-reg = <0xdc 28>;
-					clk-gate = <0xa4 2>;
-				};
-
-				emac_ptp_clk: emac_ptp_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&emac_ptp_free_clk>;
-					clk-gate = <0xa4 3>;
-				};
-
-				gpio_db_clk: gpio_db_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&gpio_db_free_clk>;
-					div-reg = <0xe0>;
-					clk-gate = <0xa4 4>;
-				};
-
-				sdmmc_clk: sdmmc_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&sdmmc_free_clk>;
-					fixed-divider = <4>;
-					clk-gate = <0xa4 5>;
-				};
-
-				s2f_usr1_clk: s2f_usr1_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&s2f_user1_free_clk>;
-					clk-gate = <0xa4 6>;
-				};
-
-				psi_ref_clk: psi_ref_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&psi_ref_free_clk>;
-					clk-gate = <0xa4 7>;
-				};
-
-				usb_clk: usb_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&l4_mp_clk>;
-					clk-gate = <0xa4 8>;
-				};
+		clocks {
+			cb_intosc_hs_div2_clk: cb_intosc_hs_div2_clk {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
+			};
 
-				spi_m_clk: spi_m_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&l4_main_clk>;
-					clk-gate = <0xa4 9>;
-				};
+			cb_intosc_ls_clk: cb_intosc_ls_clk {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
+			};
 
-				nand_clk: nand_clk {
-					#clock-cells = <0>;
-					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&l4_main_clk>;
-					clk-gate = <0xa4 10>;
-				};
+			f2s_free_clk: f2s_free_clk {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
+			};
 
-				qspi_clk: qspi_clk {
-					#clock-cells = <0>;
-					compatible = "fixed-clock";
-					clock-frequency = <400000000>;
-				};
+			osc1: osc1 {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
 			};
+
+			qspi_clk: qspi-clk {
+                                #clock-cells = <0>;
+                                compatible = "fixed-clock";
+                                clock-frequency = <200000000>;
+                        };
 		};
 
 		fpga_mgr: fpga-mgr@0 {
@@ -593,7 +158,7 @@
 			interrupts = <0 90 4>;
 			interrupt-names = "macirq";
 			mac-address = [00 00 00 00 00 00];
-			clocks = <&emac0_clk>;
+			clocks = <&clkmgr STRATIX10_EMAC0_CLK>;
 			clock-names = "stmmaceth";
 			resets = <&rst EMAC0_RESET>;
 			reset-names = "stmmaceth";
@@ -668,7 +233,7 @@
 			compatible = "snps,designware-i2c";
 			reg = <0xffc02800 0x100>;
 			interrupts = <0 103 4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst I2C0_RESET>;
 			status = "disabled";
 		};
@@ -679,7 +244,7 @@
 			compatible = "snps,designware-i2c";
 			reg = <0xffc02900 0x100>;
 			interrupts = <0 104 4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst I2C1_RESET>;
 			status = "disabled";
 		};
@@ -690,7 +255,7 @@
 			compatible = "snps,designware-i2c";
 			reg = <0xffc02a00 0x100>;
 			interrupts = <0 105 4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst I2C2_RESET>;
 			status = "disabled";
 		};
@@ -701,7 +266,7 @@
 			compatible = "snps,designware-i2c";
 			reg = <0xffc02b00 0x100>;
 			interrupts = <0 106 4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst I2C3_RESET>;
 			status = "disabled";
 		};
@@ -712,7 +277,7 @@
 			compatible = "snps,designware-i2c";
 			reg = <0xffc02c00 0x100>;
 			interrupts = <0 107 4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst I2C4_RESET>;
 			status = "disabled";
 		};
@@ -724,7 +289,8 @@
 			reg = <0xff808000 0x1000>;
 			interrupts = <0 96 4>;
 			fifo-depth = <0x400>;
-			clocks = <&l4_mp_clk>, <&sdmmc_clk>;
+			clocks = <&clkmgr STRATIX10_L4_MP_CLK>,
+                                 <&clkmgr STRATIX10_SDMMC_CLK>;
 			clock-names = "biu", "ciu";
 			resets = <&rst SDMMC_RESET>;
 			reset-names = "reset";
@@ -751,7 +317,7 @@
 			#dma-cells = <1>;
 			#dma-channels = <8>;
 			#dma-requests = <32>;
-			clocks = <&l4_main_clk>;
+			clocks = <&clkmgr STRATIX10_L4_MAIN_CLK>;
 			clock-names = "apb_pclk";
 		};
 
@@ -768,7 +334,7 @@
 			#size-cells = <0>;
 			reg = <0xffda4000 0x1000>;
 			interrupts = <0 99 4>;
-			clocks = <&spi_m_clk>;
+			clocks = <&clkmgr STRATIX10_L4_MAIN_CLK>;
 			resets = <&rst SPIM0_RESET>;
 			reg-io-width = <4>;
 			num-chipselect = <4>;
@@ -782,7 +348,7 @@
 			#size-cells = <0>;
 			reg = <0xffda5000 0x1000>;
 			interrupts = <0 100 4>;
-			clocks = <&spi_m_clk>;
+			clocks = <&clkmgr STRATIX10_L4_MAIN_CLK>;
 			resets = <&rst SPIM1_RESET>;
 			reg-io-width = <4>;
 			num-chipselect = <4>;
@@ -808,7 +374,7 @@
 			compatible = "snps,dw-apb-timer";
 			interrupts = <0 113 4>;
 			reg = <0xffc03000 0x100>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			clock-names = "timer";
 		};
 
@@ -816,7 +382,7 @@
 			compatible = "snps,dw-apb-timer";
 			interrupts = <0 114 4>;
 			reg = <0xffc03100 0x100>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			clock-names = "timer";
 		};
 
@@ -824,7 +390,7 @@
 			compatible = "snps,dw-apb-timer";
 			interrupts = <0 115 4>;
 			reg = <0xffd00000 0x100>;
-			clocks = <&l4_sys_free_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			clock-names = "timer";
 		};
 
@@ -832,7 +398,7 @@
 			compatible = "snps,dw-apb-timer";
 			interrupts = <0 116 4>;
 			reg = <0xffd00100 0x100>;
-			clocks = <&l4_sys_free_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			clock-names = "timer";
 		};
 
@@ -842,7 +408,7 @@
 			interrupts = <0 108 4>;
 			reg-shift = <2>;
 			reg-io-width = <4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst UART0_RESET>;
 			status = "disabled";
 		};
@@ -853,7 +419,7 @@
 			interrupts = <0 109 4>;
 			reg-shift = <2>;
 			reg-io-width = <4>;
-			clocks = <&l4_sp_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SP_CLK>;
 			resets = <&rst UART1_RESET>;
 			status = "disabled";
 		};
@@ -862,7 +428,7 @@
 			compatible = "snps,dwc2";
 			reg = <0xffb00000 0x40000>;
 			interrupts = <0 93 4>;
-			clocks = <&usb_clk>;
+			clocks = <&clkmgr STRATIX10_USB_CLK>;
 			clock-names = "otg";
 			resets = <&rst USB0_RESET>, <&rst USB0_OCP_RESET>;
 			reset-names = "dwc2", "dwc2-ecc";
@@ -873,7 +439,7 @@
 			compatible = "snps,dwc2";
 			reg = <0xffb40000 0x40000>;
 			interrupts = <0 94 4>;
-			clocks = <&usb_clk>;
+			clocks = <&clkmgr STRATIX10_USB_CLK>;
 			clock-names = "otg";
 			resets = <&rst USB1_RESET>, <&rst USB1_OCP_RESET>;
 			reset-names = "dwc2", "dwc2-ecc";
@@ -885,7 +451,7 @@
 			reg = <0xffd00200 0x100>;
 			interrupts = <0 117 4>;
 			resets = <&rst WATCHDOG0_RESET>;
-			clocks = <&l4_sys_free_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SYS_FREE_CLK>;
 			status = "disabled";
 		};
 
@@ -894,7 +460,7 @@
 			reg = <0xffd00300 0x100>;
 			interrupts = <0 118 4>;
 			resets = <&rst WATCHDOG1_RESET>;
-			clocks = <&l4_sys_free_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SYS_FREE_CLK>;
 			status = "disabled";
 		};
 
@@ -903,7 +469,7 @@
 			reg = <0xffd00400 0x100>;
 			interrupts = <0 125 4>;
 			resets = <&rst WATCHDOG2_RESET>;
-			clocks = <&l4_sys_free_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SYS_FREE_CLK>;
 			status = "disabled";
 		};
 
@@ -912,7 +478,7 @@
 			reg = <0xffd00500 0x100>;
 			interrupts = <0 126 4>;
 			resets = <&rst WATCHDOG3_RESET>;
-			clocks = <&l4_sys_free_clk>;
+			clocks = <&clkmgr STRATIX10_L4_SYS_FREE_CLK>;
 			status = "disabled";
 		};
 
diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
index ae99c65..9fe9c54 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
@@ -60,11 +60,9 @@
 	};
 
 	soc {
-		clkmgr@ffd10000 {
-			clocks {
-				osc1 {
-					clock-frequency = <25000000>;
-				};
+		clocks {
+			osc1 {
+				clock-frequency = <25000000>;
 			};
 		};
 
-- 
1.7.9.5

